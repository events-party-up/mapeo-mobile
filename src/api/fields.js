// @flow
import { jsonRequest } from './base';
import { computeField } from '../models/fields';

class Field {
  static list = () =>
    jsonRequest({
      method: 'GET',
      route: '/presets'
    }).map(presets => presets);
}

export default Field;
