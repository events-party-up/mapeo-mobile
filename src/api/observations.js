// @flow
import { jsonRequest } from './base';
import { parseObservationResponse } from '../models/observations';

export type ObservationAPI = {
  lat: number,
  lon: number,
  attachments: Array<{ id: string }>,
  tags: Object
};

export type UpdateRequest = {
  id: string
};

class Observation {
  static list = () =>
    jsonRequest({
      method: 'GET',
      route: '/observations'
    }).map(observations => observations.map(parseObservationResponse));

  static create = (observation: ObservationAPI) =>
    jsonRequest({
      method: 'POST',
      route: '/observations',
      body: observation
    }).map(parseObservationResponse);

  static update = (observation: ObservationAPI & UpdateRequest) =>
    jsonRequest({
      method: 'PUT',
      route: `/observations/${observation.id}`,
      body: observation
    }).map(parseObservationResponse);
}

export default Observation;
