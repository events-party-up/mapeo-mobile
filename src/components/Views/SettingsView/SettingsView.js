// @flow
import React from 'react';
import {
  StyleSheet,
  Text,
  View,
  TouchableOpacity,
  Picker,
  Dimensions
} from 'react-native';
import type { NavigationScreenProp } from 'react-navigation';
import I18n from 'react-native-i18n';
import LeftChevron from 'react-native-vector-icons/Feather';
import { LIGHT_GREY, WHITE, BLACK } from '../../../lib/styles';
import type { GPSFormat } from '../../../types/gps';
import type { Style } from '../../../types/map';

type Props = {
  navigation: NavigationScreenProp<*>
};

export type StateProps = {
  gpsFormat: GPSFormat,
  selectedStyle?: Style,
  styles: { [id: string]: Style },
  presets: string[],
  selectedPreset: string
};

export type DispatchProps = {
  setGPSFormat: (format: GPSFormat) => void,
  setSelectedStyle: (style: Style) => void,
  setSelectedPreset: (preset: string) => void
};

const styles = StyleSheet.create({
  closeDrawerButton: {
    backgroundColor: '#333333',
    borderBottomLeftRadius: 15,
    borderTopLeftRadius: 15,
    width: 40,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center'
  },
  header: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-around',
    height: 65,
    borderBottomColor: LIGHT_GREY,
    borderBottomWidth: 1,
    width: Dimensions.get('window').width,
    paddingHorizontal: 15
  },
  leftChevron: {
    alignSelf: 'center',
    justifyContent: 'center'
  },
  title: {
    fontSize: 20,
    fontWeight: '700',
    color: BLACK,
    flex: 1,
    textAlign: 'center'
  },
  empty: {
    width: 30
  },
  container: {
    flex: 1
  },
  fieldContainer: {
    padding: 20,
    flexDirection: 'column'
  },
  heading: {
    fontSize: 14,
    color: BLACK,
    marginBottom: 10,
    fontWeight: '700'
  },
  pickerContainer: {
    backgroundColor: WHITE,
    borderColor: LIGHT_GREY,
    borderWidth: 1,
    borderRadius: 3
  }
});

I18n.fallbacks = true;
I18n.translations = {
  en: require('../../../translations/en'),
  es: require('../../../translations/es')
};

class SettingsView extends React.PureComponent<
  Props & StateProps & DispatchProps
> {
  handleValueSelect = (format: GPSFormat) => {
    const { setGPSFormat } = this.props;

    setGPSFormat(format);
  };

  handleStyleSelect = (style: string) => {
    const { setSelectedStyle, styles } = this.props;

    setSelectedStyle(styles[style]);
  };

  handlePresetSelect = (preset: string) => {
    const { setSelectedPreset } = this.props;

    setSelectedPreset(preset);
  };

  render() {
    const {
      gpsFormat,
      navigation,
      selectedStyle,
      selectedPreset,
      presets
    } = this.props;
    const mapStyles = this.props.styles;

    return (
      <View
        style={{
          flex: 1,
          flexDirection: 'column',
          backgroundColor: WHITE,
          justifyContent: 'flex-start'
        }}
      >
        <View style={styles.header}>
          <TouchableOpacity
            onPress={() => {
              navigation.goBack();
            }}
          >
            <LeftChevron color="#a5a5a4" name="chevron-left" size={30} />
          </TouchableOpacity>
          <Text style={styles.title}>{I18n.t('settings.settings')}</Text>
          <View style={styles.empty} />
        </View>
        <View style={styles.container}>
          <View style={styles.fieldContainer}>
            <Text style={styles.heading}>{I18n.t('settings.gps_format')}</Text>
            <View style={styles.pickerContainer}>
              <Picker
                mode="dropdown"
                itemStyle={{ width: Dimensions.get('window').width - 40 }}
                onValueChange={this.handleValueSelect}
                selectedValue={gpsFormat}
              >
                <Picker.Item
                  label={`${I18n.t('settings.dd')} (DDD.DDDDD°)`}
                  value="DD"
                />
                <Picker.Item
                  label={`${I18n.t('settings.ddm')} (DDD° MM.MMM')`}
                  value="DDM"
                />
                <Picker.Item
                  label={`${I18n.t('settings.dms')} (DDD° MM' SS.S")`}
                  value="DMS"
                />
                <Picker.Item
                  label={`${I18n.t('settings.utm')} (10S 123456m E 7654321m N)`}
                  value="UTM"
                />
              </Picker>
            </View>
          </View>
          <View style={styles.fieldContainer}>
            <Text style={styles.heading}>{I18n.t('settings.map_style')}</Text>
            <View style={styles.pickerContainer}>
              <Picker
                mode="dropdown"
                itemStyle={{ width: Dimensions.get('window').width - 40 }}
                onValueChange={this.handleStyleSelect}
                selectedValue={selectedStyle && selectedStyle.id}
              >
                {Object.keys(mapStyles).map(id => (
                  <Picker.Item
                    label={mapStyles[id].name.toString()}
                    value={id}
                    key={id}
                  />
                ))}
              </Picker>
            </View>
          </View>
          <View style={styles.fieldContainer}>
            <Text style={styles.heading}>{I18n.t('settings.presets')}</Text>
            <View style={styles.pickerContainer}>
              <Picker
                mode="dropdown"
                itemStyle={{ width: Dimensions.get('window').width - 40 }}
                onValueChange={this.handlePresetSelect}
                selectedValue={selectedPreset}
              >
                {presets.map(p => <Picker.Item label={p} value={p} key={p} />)}
              </Picker>
            </View>
          </View>
        </View>
      </View>
    );
  }
}

export default SettingsView;
