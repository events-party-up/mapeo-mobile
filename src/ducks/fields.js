// @flow

import update from 'immutability-helper';
import { keyBy } from 'lodash';
import { create } from '../lib/redux';

export const {
  type: FIELD_LIST,
  action: fieldList,
  reducer: fieldListReducer
} = create('FIELD_LIST', {
  success: (state, meta, payload) => {
    const newState = update(state, {
      fields: { $set: keyBy(payload, 'id') }
    });

    return newState;
  }
});

export default [fieldListReducer];
