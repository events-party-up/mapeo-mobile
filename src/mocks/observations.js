// @flow

import { createField } from '../mocks/fields';
import type { Observation } from '../types/observation';

export const createObservation = (observation?: any): Observation => ({
  id: 1,
  lat: 50.5,
  lon: 50.5,
  created: new Date('2018-02-25T20:23:00.000Z'),
  name: 'Oil Spill',
  notes:
    'This is pretty bad. It smells pretty noxious, and the entire pond is affected.',
  observedBy: 'You',
  attachments: [],
  categoryId: 19,
  fields: [createField()],

  ...observation
});
