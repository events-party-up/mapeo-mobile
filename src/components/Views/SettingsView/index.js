// @flow
import { connect } from 'react-redux';
import type { Dispatch } from 'redux';
import SettingsView from './SettingsView';
import type { StoreState } from '../../../types/redux';
import type { StateProps, DispatchProps } from './SettingsView';
import { gpsFormatSettingsSet } from '../../../ducks/settings';
import { styleSelect } from '../../../ducks/map';
import { presetsSelect } from '../../../ducks/presets';

function mapStateToProps(state: StoreState): StateProps {
  return {
    gpsFormat: state.settings.gpsFormat,
    selectedStyle: state.map.selectedStyle,
    styles: state.map.styles,
    presets: state.presets,
    selectedPreset: state.selectedPreset
  };
}

function mapDispatchToProps(dispatch: Dispatch<*>) {
  return {
    setGPSFormat: format => dispatch(gpsFormatSettingsSet(format)),
    setSelectedStyle: style => dispatch(styleSelect(style)),
    setSelectedPreset: preset => dispatch(presetsSelect(preset))
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(SettingsView);
