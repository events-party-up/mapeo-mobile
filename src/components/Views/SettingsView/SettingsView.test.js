// @flow
import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import SettingsView from './SettingsView';
import type { StateProps } from './SettingsView';
import { createObservation } from '../../../mocks/observations';
import { createField } from '../../../mocks/fields';
import { createStyle } from '../../../mocks/map';
import { createNavigationScreenProp } from '../../../mocks/navigation';

describe('SettingsView tests', () => {
  const setGPSFormat = jest.fn();
  const setSelectedStyle = jest.fn();
  const setSelectedPreset = jest.fn();
  const style = createStyle();
  const styles = {
    [style.id]: style
  };
  const presets = ['presets0', 'presets1'];

  beforeEach(() => {
    setGPSFormat.mockReset();
    setSelectedStyle.mockReset();
    setSelectedPreset.mockReset();
  });

  test('snapshot', () => {
    const props: StateProps[] = [
      { gpsFormat: 'DD', styles, presets, selectedPreset: presets[0] },
      { gpsFormat: 'DDM', styles, presets, selectedPreset: presets[0] },
      { gpsFormat: 'DMS', styles, presets, selectedPreset: presets[0] },
      { gpsFormat: 'UTM', styles, presets, selectedPreset: presets[0] }
    ];

    let tree;
    props.forEach(p => {
      tree = renderer.create(
        <SettingsView
          navigation={createNavigationScreenProp()}
          gpsFormat={p.gpsFormat}
          setGPSFormat={setGPSFormat}
          setSelectedStyle={setSelectedStyle}
          styles={p.styles}
          presets={p.presets}
          selectedPreset={p.selectedPreset}
          setSelectedPreset={setSelectedPreset}
        />
      );
      expect(tree).toMatchSnapshot();
    });
  });
});
