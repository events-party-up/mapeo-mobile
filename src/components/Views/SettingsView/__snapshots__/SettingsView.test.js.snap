// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsView tests snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "flex-start",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "#EAEAEA",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "height": 65,
        "justifyContent": "space-around",
        "paddingHorizontal": 15,
        "width": 750,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#a5a5a4",
              "fontSize": 30,
            },
            undefined,
            Object {
              "fontFamily": "Feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000000",
          "flex": 1,
          "fontSize": 20,
          "fontWeight": "700",
          "textAlign": "center",
        }
      }
    >
      Settings
    </Text>
    <View
      style={
        Object {
          "width": 30,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "column",
          "padding": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
            "fontSize": 14,
            "fontWeight": "700",
            "marginBottom": 10,
          }
        }
      >
        GPS Coordinate Format
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderColor": "#EAEAEA",
            "borderRadius": 3,
            "borderWidth": 1,
          }
        }
      >
        <View>
          <RCTPicker
            items={
              Array [
                Object {
                  "label": "Decimal Degrees (DDD.DDDDD°)",
                  "textColor": undefined,
                  "value": "DD",
                },
                Object {
                  "label": "Degrees & Decimal Minutes (DDD° MM.MMM')",
                  "textColor": undefined,
                  "value": "DDM",
                },
                Object {
                  "label": "Degrees, Minutes, Seconds (DDD° MM' SS.S\\")",
                  "textColor": undefined,
                  "value": "DMS",
                },
                Object {
                  "label": "UTM Coordinates (10S 123456m E 7654321m N)",
                  "textColor": undefined,
                  "value": "UTM",
                },
              ]
            }
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            selectedIndex={0}
            style={
              Array [
                Object {
                  "height": 216,
                },
                Object {
                  "width": 710,
                },
              ]
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "column",
          "padding": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
            "fontSize": 14,
            "fontWeight": "700",
            "marginBottom": 10,
          }
        }
      >
        Map Style
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderColor": "#EAEAEA",
            "borderRadius": 3,
            "borderWidth": 1,
          }
        }
      >
        <View>
          <RCTPicker
            items={
              Array [
                Object {
                  "label": "Test Style",
                  "textColor": undefined,
                  "value": "test-style",
                },
              ]
            }
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            selectedIndex={0}
            style={
              Array [
                Object {
                  "height": 216,
                },
                Object {
                  "width": 710,
                },
              ]
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "column",
          "padding": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
            "fontSize": 14,
            "fontWeight": "700",
            "marginBottom": 10,
          }
        }
      >
        Presets
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderColor": "#EAEAEA",
            "borderRadius": 3,
            "borderWidth": 1,
          }
        }
      >
        <View>
          <RCTPicker
            items={
              Array [
                Object {
                  "label": "presets0",
                  "textColor": undefined,
                  "value": "presets0",
                },
                Object {
                  "label": "presets1",
                  "textColor": undefined,
                  "value": "presets1",
                },
              ]
            }
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            selectedIndex={0}
            style={
              Array [
                Object {
                  "height": 216,
                },
                Object {
                  "width": 710,
                },
              ]
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`SettingsView tests snapshot 2`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "flex-start",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "#EAEAEA",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "height": 65,
        "justifyContent": "space-around",
        "paddingHorizontal": 15,
        "width": 750,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#a5a5a4",
              "fontSize": 30,
            },
            undefined,
            Object {
              "fontFamily": "Feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000000",
          "flex": 1,
          "fontSize": 20,
          "fontWeight": "700",
          "textAlign": "center",
        }
      }
    >
      Settings
    </Text>
    <View
      style={
        Object {
          "width": 30,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "column",
          "padding": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
            "fontSize": 14,
            "fontWeight": "700",
            "marginBottom": 10,
          }
        }
      >
        GPS Coordinate Format
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderColor": "#EAEAEA",
            "borderRadius": 3,
            "borderWidth": 1,
          }
        }
      >
        <View>
          <RCTPicker
            items={
              Array [
                Object {
                  "label": "Decimal Degrees (DDD.DDDDD°)",
                  "textColor": undefined,
                  "value": "DD",
                },
                Object {
                  "label": "Degrees & Decimal Minutes (DDD° MM.MMM')",
                  "textColor": undefined,
                  "value": "DDM",
                },
                Object {
                  "label": "Degrees, Minutes, Seconds (DDD° MM' SS.S\\")",
                  "textColor": undefined,
                  "value": "DMS",
                },
                Object {
                  "label": "UTM Coordinates (10S 123456m E 7654321m N)",
                  "textColor": undefined,
                  "value": "UTM",
                },
              ]
            }
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            selectedIndex={1}
            style={
              Array [
                Object {
                  "height": 216,
                },
                Object {
                  "width": 710,
                },
              ]
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "column",
          "padding": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
            "fontSize": 14,
            "fontWeight": "700",
            "marginBottom": 10,
          }
        }
      >
        Map Style
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderColor": "#EAEAEA",
            "borderRadius": 3,
            "borderWidth": 1,
          }
        }
      >
        <View>
          <RCTPicker
            items={
              Array [
                Object {
                  "label": "Test Style",
                  "textColor": undefined,
                  "value": "test-style",
                },
              ]
            }
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            selectedIndex={0}
            style={
              Array [
                Object {
                  "height": 216,
                },
                Object {
                  "width": 710,
                },
              ]
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "column",
          "padding": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
            "fontSize": 14,
            "fontWeight": "700",
            "marginBottom": 10,
          }
        }
      >
        Presets
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderColor": "#EAEAEA",
            "borderRadius": 3,
            "borderWidth": 1,
          }
        }
      >
        <View>
          <RCTPicker
            items={
              Array [
                Object {
                  "label": "presets0",
                  "textColor": undefined,
                  "value": "presets0",
                },
                Object {
                  "label": "presets1",
                  "textColor": undefined,
                  "value": "presets1",
                },
              ]
            }
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            selectedIndex={0}
            style={
              Array [
                Object {
                  "height": 216,
                },
                Object {
                  "width": 710,
                },
              ]
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`SettingsView tests snapshot 3`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "flex-start",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "#EAEAEA",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "height": 65,
        "justifyContent": "space-around",
        "paddingHorizontal": 15,
        "width": 750,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#a5a5a4",
              "fontSize": 30,
            },
            undefined,
            Object {
              "fontFamily": "Feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000000",
          "flex": 1,
          "fontSize": 20,
          "fontWeight": "700",
          "textAlign": "center",
        }
      }
    >
      Settings
    </Text>
    <View
      style={
        Object {
          "width": 30,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "column",
          "padding": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
            "fontSize": 14,
            "fontWeight": "700",
            "marginBottom": 10,
          }
        }
      >
        GPS Coordinate Format
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderColor": "#EAEAEA",
            "borderRadius": 3,
            "borderWidth": 1,
          }
        }
      >
        <View>
          <RCTPicker
            items={
              Array [
                Object {
                  "label": "Decimal Degrees (DDD.DDDDD°)",
                  "textColor": undefined,
                  "value": "DD",
                },
                Object {
                  "label": "Degrees & Decimal Minutes (DDD° MM.MMM')",
                  "textColor": undefined,
                  "value": "DDM",
                },
                Object {
                  "label": "Degrees, Minutes, Seconds (DDD° MM' SS.S\\")",
                  "textColor": undefined,
                  "value": "DMS",
                },
                Object {
                  "label": "UTM Coordinates (10S 123456m E 7654321m N)",
                  "textColor": undefined,
                  "value": "UTM",
                },
              ]
            }
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            selectedIndex={2}
            style={
              Array [
                Object {
                  "height": 216,
                },
                Object {
                  "width": 710,
                },
              ]
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "column",
          "padding": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
            "fontSize": 14,
            "fontWeight": "700",
            "marginBottom": 10,
          }
        }
      >
        Map Style
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderColor": "#EAEAEA",
            "borderRadius": 3,
            "borderWidth": 1,
          }
        }
      >
        <View>
          <RCTPicker
            items={
              Array [
                Object {
                  "label": "Test Style",
                  "textColor": undefined,
                  "value": "test-style",
                },
              ]
            }
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            selectedIndex={0}
            style={
              Array [
                Object {
                  "height": 216,
                },
                Object {
                  "width": 710,
                },
              ]
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "column",
          "padding": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
            "fontSize": 14,
            "fontWeight": "700",
            "marginBottom": 10,
          }
        }
      >
        Presets
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderColor": "#EAEAEA",
            "borderRadius": 3,
            "borderWidth": 1,
          }
        }
      >
        <View>
          <RCTPicker
            items={
              Array [
                Object {
                  "label": "presets0",
                  "textColor": undefined,
                  "value": "presets0",
                },
                Object {
                  "label": "presets1",
                  "textColor": undefined,
                  "value": "presets1",
                },
              ]
            }
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            selectedIndex={0}
            style={
              Array [
                Object {
                  "height": 216,
                },
                Object {
                  "width": 710,
                },
              ]
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`SettingsView tests snapshot 4`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "flex-start",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "#EAEAEA",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "height": 65,
        "justifyContent": "space-around",
        "paddingHorizontal": 15,
        "width": 750,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#a5a5a4",
              "fontSize": 30,
            },
            undefined,
            Object {
              "fontFamily": "Feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000000",
          "flex": 1,
          "fontSize": 20,
          "fontWeight": "700",
          "textAlign": "center",
        }
      }
    >
      Settings
    </Text>
    <View
      style={
        Object {
          "width": 30,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "column",
          "padding": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
            "fontSize": 14,
            "fontWeight": "700",
            "marginBottom": 10,
          }
        }
      >
        GPS Coordinate Format
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderColor": "#EAEAEA",
            "borderRadius": 3,
            "borderWidth": 1,
          }
        }
      >
        <View>
          <RCTPicker
            items={
              Array [
                Object {
                  "label": "Decimal Degrees (DDD.DDDDD°)",
                  "textColor": undefined,
                  "value": "DD",
                },
                Object {
                  "label": "Degrees & Decimal Minutes (DDD° MM.MMM')",
                  "textColor": undefined,
                  "value": "DDM",
                },
                Object {
                  "label": "Degrees, Minutes, Seconds (DDD° MM' SS.S\\")",
                  "textColor": undefined,
                  "value": "DMS",
                },
                Object {
                  "label": "UTM Coordinates (10S 123456m E 7654321m N)",
                  "textColor": undefined,
                  "value": "UTM",
                },
              ]
            }
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            selectedIndex={3}
            style={
              Array [
                Object {
                  "height": 216,
                },
                Object {
                  "width": 710,
                },
              ]
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "column",
          "padding": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
            "fontSize": 14,
            "fontWeight": "700",
            "marginBottom": 10,
          }
        }
      >
        Map Style
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderColor": "#EAEAEA",
            "borderRadius": 3,
            "borderWidth": 1,
          }
        }
      >
        <View>
          <RCTPicker
            items={
              Array [
                Object {
                  "label": "Test Style",
                  "textColor": undefined,
                  "value": "test-style",
                },
              ]
            }
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            selectedIndex={0}
            style={
              Array [
                Object {
                  "height": 216,
                },
                Object {
                  "width": 710,
                },
              ]
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "column",
          "padding": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
            "fontSize": 14,
            "fontWeight": "700",
            "marginBottom": 10,
          }
        }
      >
        Presets
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderColor": "#EAEAEA",
            "borderRadius": 3,
            "borderWidth": 1,
          }
        }
      >
        <View>
          <RCTPicker
            items={
              Array [
                Object {
                  "label": "presets0",
                  "textColor": undefined,
                  "value": "presets0",
                },
                Object {
                  "label": "presets1",
                  "textColor": undefined,
                  "value": "presets1",
                },
              ]
            }
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            selectedIndex={0}
            style={
              Array [
                Object {
                  "height": 216,
                },
                Object {
                  "width": 710,
                },
              ]
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
